ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js

```diff
@@ -0,0 +1,86 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Identicon from '../../../../components/ui/identicon'
+import { CONTACT_LIST_ROUTE, CONTACT_EDIT_ROUTE } from '../../../../helpers/constants/routes'
+import { addressSlicer } from '../../../../helpers/utils/util'
+import Button from '../../../../components/ui/button/button.component'
+const copyToClipboard = require('copy-to-clipboard')
+
+export default class ViewContact extends PureComponent {
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    addressBook: PropTypes.object,
+    addToAddressBook: PropTypes.func,
+    removeFromAddressBook: PropTypes.func,
+    history: PropTypes.object,
+    match: PropTypes.object,
+  }
+
+   render () {
+    const { t } = this.context
+    const { removeFromAddressBook, history, match, addressBook } = this.props
+    const address = match.params.id
```

Comment:
A pattern (that requires documentation!) that we try to follow is to, as much as possible, keep logic related to the selection or modification of data in the `container.js` file.

I suggest moving lines 26-28 to the container, and just passing `name` to the component.

In case you are not aware, `mapStateToProps` always receives a second param, usually referred to as `ownProps`. This will include any properties passed by the parent. When using `withRouter` as done in `ui/app/pages/settings/contact-list-tab/view-contact/view-contact.container.js`, it will also include the `match` property.